app/scripts/modules/core/interceptor/debug.interceptor.ts

```diff
@@ -0,0 +1,32 @@
+import {module, IRequestConfig, IHttpInterceptor, IHttpProvider} from 'angular';
+import {$log, $location} from 'ngimport';
+import {JSON_UTILITY_SERVICE, JsonUtilityService} from 'core/utils/json/json.utility.service';
+import autoBindMethods from 'class-autobind-decorator';
+
+@autoBindMethods
```

Comment:
`request` calls the other class method (and interceptors lose their `this` bindings). It could all be in one method to keep it simple.